packages/shared/src/hooks/post/usePostActions.ts

```diff
@@ -20,15 +21,20 @@ export const usePostActions = ({ post }: { post: Post }): UsePostActions => {
     return generateQueryKey(RequestKey.PostActions, { id: post?.id });
   }, [post?.id]);
 
-  const { data } = useQuery<PostActionData>({
+  const queryFn = useCallback((): PostActionData => {
+    return {
+      interaction: 'none',
+      previousInteraction: 'none',
+    };
+  }, []);
+
+  const { data } = useQuery({
     queryKey: key,
-    initialData: () => {
-      return {
-        interaction: 'none',
-        previousInteraction: 'none',
-      };
-    },
+    queryFn,
+    initialData: queryFn,
     staleTime: Infinity,
+    gcTime: Infinity,
+    ...disabledRefetch,
```

Comment:
Basically:
- use initial data as queryFn
- disable refetch
- cache Infinite

Can still be manipulated through queryClient